flink-ml-python/pyflink/ml/recommendation/tests/test_swing.py

```diff
@@ -0,0 +1,149 @@
+from pyflink.table import Table
+from typing import List
+from py4j.protocol import Py4JJavaError:
+
+from pyflink.ml.recommendation.swing import Swing
+from pyflink.ml.tests.test_utils import PyFlinkMLTestCase
+
+
+# Tests Swing. 
+class SwingTest(PyFlinkMLTestCase):
+    def setUp(self):
+        super(SwingTest, self).setUp()
+        self.train_data = self.t_env.from_data_stream(
+            self.env.from_collection([
+                (0, 10),
+                (0, 11),
+                (0, 12),
+                (1, 13),
+                (1, 12),
+                (2, 10),
+                (2, 11),
+                (2, 12),
+                (3, 13),
+                (3, 12)
+            ],
+                type_info=Types.ROW_NAMED(
+                    ['user', 'item'],
+                    [Types.LONG(), Types.LONG()])
+            ))
+
+        self.wrong_type_data = self.t_env.from_data_stream(
+            self.env.from_collection([
+                (0, 10),
+                (1, 11),
+                (2, 12)
+            ],
+                type_info=Types.ROW_NAMED(
+                    ['user', 'item'],
+                    [Types.INT(), Types.LONG()])
+            ))
+
+        self.none_value_data = self.t_env.from_data_stream(
+            self.env.from_collection([
+                (0, 10),
+                (None, 11),
+                (2, 12)
+            ],
+                type_info=Types.ROW_NAMED(
+                    ['user', 'item'],
+                    [Types.LONG(), Types.LONG()])
+            ))
+
+        self.expected_data = [
+            [10, '11,0.058845768947156235;12,0.058845768947156235'],
+            [11, '10,0.058845768947156235;12,0.058845768947156235'],
+            [12, '13,0.09134833828228624;10,0.058845768947156235;11,0.058845768947156235'],
+            [13, '12,0.09134833828228624']]
+
+    def test_param(self):
+        swing = Swing()
+        self.assertEqual("item", swing.item_col)
+        self.assertEqual("user", swing.user_col)
+        self.assertEqual(100, swing.k)
+        self.assertEqual(10, swing.min_user_items)
+        self.assertEqual(1000, swing.max_user_items)
+        self.assertEqual(15, swing.alpha1)
+        self.assertEqual(0, swing.alpha2)
+        self.assertEqual(0.3, swing.beta, delta=1e-9)
+
+        swing.set_item_col("item_1") \
+            .set_user_col("user_1") \
+            .set_k(20) \
+            .set_min_user_behavior(20) \
+            .set_max_user_behavior(50) \
+            .set_alpha1(5) \
+            .set_alpha2(1) \
+            .set_beta(0.35)
+
+        self.assertEqual("item", swing.item_col)
+        self.assertEqual("user", swing.user_col)
+        self.assertEqual(20, swing.k)
+        self.assertEqual(20, swing.min_user_items)
+        self.assertEqual(50, swing.max_user_items)
+        self.assertEqual(5, swing.alpha1)
+        self.assertEqual(1, swing.alpha2)
+        self.assertEqual(0.35, swing.beta, delta=1e-9)
```

Comment:
In Python, we should use `assertAlmostEqual`. The IDE should give warning about this line.